### DISCUSSION TITLE
Is there any predefined search widget?

### DISCUSSION THREAD

**Discussion by MohamedElgamal** [Textualize/textual#3227] [General]
Looking for following widget that has this specifications look like select widget in design but need to add ability of typing in that widget and when user typing it show in the same type the closest word for what user type in the widget like i am searching for specific username and when typing in the widget it filters and show if that widget exists or not, if not is anyone have suggestions?

**Comment by davep** [comment]
Would [this library](https://github.com/darrenburns/textual-autocomplete) be close to what you're looking for?

**Comment by MohamedElgamal** [comment]
yes, but there is something different the DropDownItem will not be determined in compile time it will be determined in runtime because database will decide what list to return to user based on what character he typing if he type "ha" it will return all users that contain "ha" characters

**Reply by davep** [reply]
Sounds like [this part of the README](https://github.com/darrenburns/textual-autocomplete#using-a-callable) addresses that use case, perhaps?
